src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -97,18 +97,37 @@ struct
                    , `Staged_ledger transitioned_staged_ledger
                    , `Pending_coinbase_data _ ) =
             let open Deferred.Let_syntax in
-            match%map
+            match%bind
               Staged_ledger.apply ~logger staged_ledger
                 (External_transition.Verified.staged_ledger_diff transition)
             with
             | Ok x ->
-                Ok x
+                return (Ok x)
             | Error (Staged_ledger.Staged_ledger_error.Unexpected e) ->
-                Error (`Fatal_error (Error.to_exn e))
-            | Error e ->
-                Error
-                  (`Invalid_staged_ledger_diff
-                    (Staged_ledger.Staged_ledger_error.to_error e))
+                return (Error (`Fatal_error (Error.to_exn e)))
+            | Error staged_ledger_error ->
+                let%bind () =
+                  match sender with
+                  | None | Some Envelope.Sender.Local ->
+                      return ()
+                  | Some (Envelope.Sender.Remote inet_addr) ->
+                      let error_string =
+                        Staged_ledger.Staged_ledger_error.to_string
+                          staged_ledger_error
+                      in
+                      Trust_system.(
+                        record trust_system logger inet_addr
+                          Actions.
+                            ( Violated_protocol
+                            , Some
+                                ( "Staged_ledger error: $error"
```

Comment:
This failure is recorded twice, first here as `Violated_protocol` and then as `Gossiped_invalid_transition` in `processor.ml`. Shouldn't it be recorded just once?

Reply:
removed this use of the trust system

distinguished errors in build